File: app/src/main/java/net/tlalka/android/fiszki/view/activities/WelcomeActivity.java
package net.tlalka.android.fiszki.view.activities;

import javax.inject.Inject;

import android.os.Bundle;
import android.view.View;
import android.widget.TextView;

import butterknife.BindView;
import butterknife.OnClick;
import net.tlalka.android.fiszki.R;
import net.tlalka.android.fiszki.model.dto.WelcomeDto;
import net.tlalka.android.fiszki.view.navigations.Navigator;

public class WelcomeActivity extends AbstractActivity {

    @BindView(R.id.welcome_message)
    protected TextView message;

    @Inject
    protected Navigator navigator;

    @Inject
    protected WelcomeDto welcomeDto;

    @Override
    public void onCreate(Bundle bundle) {
        super.onCreate(bundle);
        super.setContentView(R.layout.welcome_activity);
        super.getActivityComponent().inject(this);

        this.message.setText(welcomeDto.getMessage());
    }

    @OnClick(R.id.welcome_layout)
    public void onLayoutClick(View view) {
        this.navigator.finish(this);
    }
}

File: app/src/main/java/net/tlalka/android/fiszki/view/activities/WelcomeActivity.kt
package net.tlalka.android.fiszki.view.activities

import javax.inject.Inject

import android.os.Bundle
import android.view.View
import android.widget.TextView

import butterknife.BindView
import butterknife.OnClick
import net.tlalka.android.fiszki.R
import net.tlalka.android.fiszki.model.dto.WelcomeDto
import net.tlalka.android.fiszki.view.navigations.Navigator

class WelcomeActivity : AbstractActivity() {

    @BindView(R.id.welcome_message)
    lateinit var message: TextView

    @Inject
    lateinit var navigator: Navigator

    @Inject
    lateinit var welcomeDto: WelcomeDto

    public override fun onCreate(bundle: Bundle?) {
        super.onCreate(bundle)
        super.setContentView(R.layout.welcome_activity)
        super.activityComponent.inject(this)

        message.text = welcomeDto.message
    }

    @OnClick(R.id.welcome_layout)
    fun onLayoutClick(view: View) {
        navigator.finish(this)
    }
}
